src/omnicore/tx.cpp

```diff
@@ -1869,6 +1881,9 @@ int CMPTransaction::logicMath_RevokeTokens()
     assert(update_tally_map(sender, property, -nValue, BALANCE));
     assert(_my_sps->updateSP(property, sp));
 
+    // Number of tokens has changed, update fee distribution thresholds
+    p_feecache->UpdateDistributionThresholds();
```

Comment:
Hmm.. I don't really like that logic functions now also "have to care" about updating the fee threshold, but I see how this is much more targeted than checking the thresholds every time.